rfc/0000-keyless-job-auth.rst

```diff
@@ -0,0 +1,204 @@
+   |------------------------------------------------------------------------------------------------------------------------------------------------------------------->|
+   |                                                                                        |                                                      |                    | -----------------------------------\
+   |                                                                                        |                                                      |                    |-| Validate access token and scopes |
+   |                                                                                        |                                                      |                    | |----------------------------------|
+   |                                                                                        |                                                      |                    |
+   |                                                                                        |                                                      |               File |
+   |<-------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+   |                                                                                        |                                                      |                                      |
+
+
+It is worth emphasizing that the purpose of this feature is *not* to provide a
+fully complete and compliant metadata server to Hail Batch. Rather, the aim is to provide the
+minimum functionality necessary to allow Hail libraries and popular first-party
+tools like `gcloud` and `az` the ability to obtain short-lived credentials without
+exposing key files to user code. As such, an implementation may implement just the
+endpoints necessary to run the below examples for at least one version of `gcloud`/`az`
+and all supported versions of `hail`.
+
+
+Examples
+--------
+
+Under the proposed change, the following Batch job commands should succeed:
+
+.. code-block:: python
+   j.command('gcloud storage ls <MY_BUCKET>')
+   j.command('hailctl batch submit <MY_SCRIPT>')
+
+
+Effect and Interactions
+-----------------------
+
+This change adds a method through which jobs can obtained short-lived access
+tokens without directly accessing long-lived credentials. With such a solution
+in place, we can eventually remove the long-lived credentials from job containers,
+mitigating the risk of exfiltration.
+
+So long as the firewall rules are correctly configured to redirect metadata traffic
+to the Batch worker, there should be no adverse interactions with the existing
+system as such traffic was previously forbidden.
```

Comment:
Worth noting here that anyone who has been either explicitly reading the `/gsa-key/key.json` or explicitly using the environment variable will break once we remove the key mounts (which we really should do post-haste b/c they are a bit of a security nightmare).

I'm sure there's at least one user out there who is doing this explicitly in some Python script or whatever that they use in a Batch job. AFAIK, it should always be as simple as switching to the language-appropriate version of  `google.auth.default()`.